diesel/src/query_dsl/save_changes_dsl.rs

```diff
@@ -7,26 +7,33 @@ use query_dsl::{LoadQuery, RunQueryDsl};
 use result::QueryResult;
 #[cfg(any(feature = "sqlite", feature = "mysql"))]
 use associations::Identifiable;
+use connection::Connection;
 #[cfg(any(feature = "sqlite", feature = "mysql"))]
 use dsl::Find;
 #[cfg(any(feature = "sqlite", feature = "mysql"))]
 use query_dsl::methods::{ExecuteDsl, FindDsl};
 
-#[doc(hidden)]
-pub trait InternalSaveChangesDsl<T, U> {
-    fn internal_save_changes(&self, changeset: T) -> QueryResult<U>;
+/// A trait defining how to update a record and fetch the updated entry
+/// on a certain backend.
+///
+/// The only case where it is required to work with this trait is while
+/// implementing a new connection type.
+/// Otherwise use [`SaveChangesDsl`](trait.SaveChangesDsl.html)
+pub trait UpdateAndFetchResults<T, U>: Connection {
+    /// See the traits documentation.
```

Comment:
I think, you can describe, than usually there will be changeset and what are `T` and `U`